ocaml/typing/oprint.ml

```diff
@@ -148,6 +148,34 @@ let rec print_typlist print_elem sep ppf =
       pp_print_space ppf ();
       print_typlist print_elem sep ppf tyl
 
+let print_label_type ppf =
+  function
+  | Some s ->
+    pp_print_string ppf s;
+    pp_print_string ppf ": ";
+  | None -> ()
+    
+let print_label ppf =
+  function
+  | Some s ->
+    pp_print_string ppf "~";
+    pp_print_string ppf s;
+    pp_print_string ppf ": ";
+  | None -> ()
+
+let rec print_labeled_typlist print_elem sep ppf =
+  function
+    [] -> ()
+  | [label, ty] ->
+      print_label_type ppf label;
+      print_elem ppf ty
+  | (label, ty) :: tyl ->
+      print_label_type ppf label;
+      print_elem ppf ty;
+      pp_print_string ppf sep;
+      pp_print_space ppf ();
+      print_labeled_typlist print_elem sep ppf tyl
```

Comment:
It's tempting to try to reuse part of the printing of labeled function types or record types (e.g., `print_out_label`).  I'm not sure how much that saves us, though.

Either way, there are a couple things to clean up here:
- Printing a space only on the right of the colon looks a little off to me.  I'd be fine with printing a space on both sides (like record types) or no space (like labeled function arguments).
- We should probably put a pretty printing box (`@[` and `@]`) around each element to prevent weird line splitting, and maybe use space hints instead of a literal space after the colon (if you keep the spaces).  See how this is done in `print_out_labels`, and you can read the [documentation](https://v2.ocaml.org/api/Format.html) of the Format library to get a sense of what's going on with all the weird characters (or we can discuss).